core/ecschema-editing/src/test/Merging/KindOfQuantityMerger.test.ts

```diff
@@ -0,0 +1,381 @@
+    version: "1.0.0",
+    alias: "target",
+  };
+  const referenceJson = {
+    $schema: "https://dev.bentley.com/json_schemas/ec/32/ecschema",
+    name: "ReferenceSchema",
+    version: "1.2.0",
+    alias: "reference",
+    items: {
+      TestUnitSystem: {
+        schemaItemType: "UnitSystem",
+      },
+      TestPhenomenon: {
+        schemaItemType: "Phenomenon",
+        definition: "Test Phenomenon",
+      },
+      TU_PER_TU: {
+        schemaItemType: "Unit",
+        unitSystem: "ReferenceSchema.TestUnitSystem",
+        phenomenon: "ReferenceSchema.TestPhenomenon",
+        definition: "TU",
+      },
+      TU_HORIZONTAL_PER_TU_VERTICAL: {
+        schemaItemType: "InvertedUnit",
+        invertsUnit: "ReferenceSchema.TU_PER_TU",
+        unitSystem: "ReferenceSchema.TestUnitSystem",
+      },
+      TestDecimal: {
+        schemaItemType: "Format",
+        type: "Decimal",
+        precision: 6,
+        formatTraits: [
+          "KeepSingleZero",
+          "KeepDecimalPoint",
+          "ShowUnitLabel",
+        ],
+        decimalSeparator: ",",
+        thousandSeparator: " ",
+      },
+    },
+  };
+
+  beforeEach(async () => {
+    targetContext = new SchemaContext();
+    sourceContext = new SchemaContext();
+    await Schema.fromJson(referenceJson, sourceContext);
+  });
+
+  describe("KindOfQuantity missing tests", () => {
+    it("should merge missing kind of quantity with persistent InvertedUnit ", async () => {
+      const sourceSchema = await Schema.fromJson({
+        ...sourceJson,
+        references: [
+          {
+            name: "ReferenceSchema",
+            version: "1.2.0",
+          },
+        ],
+        items: {
+          TestKoq: {
+            schemaItemType: "KindOfQuantity",
+            label: "Test",
+            description: "Description of koq",
+            relativeError: 1.23,
+            persistenceUnit: "ReferenceSchema.TU_HORIZONTAL_PER_TU_VERTICAL",
+          },
+        },
+      }, sourceContext);
+
+      const targetSchema = await Schema.fromJson({
+        ...targetJson,
+      }, targetContext);
+
+      const merger = new SchemaMerger();
+      const mergedSchema = await merger.merge(targetSchema, sourceSchema);
+      const sourceItem = await sourceSchema.getItem<KindOfQuantity>("TestKoq");
+      const mergedItem = await mergedSchema.getItem<KindOfQuantity>("TestKoq");
+      expect(mergedItem!.toJSON()).deep.eq(sourceItem!.toJSON());
+    });
+
+    it("should merge missing kind of quantity with persistent Unit", async () => {
+      const sourceSchema = await Schema.fromJson({
+        ...sourceJson,
+        references: [
+          {
+            name: "ReferenceSchema",
+            version: "1.2.0",
+          },
+        ],
+        items: {
+          TU_PER_HR: {
+            schemaItemType: "Unit",
+            unitSystem: "ReferenceSchema.TestUnitSystem",
+            phenomenon: "ReferenceSchema.TestPhenomenon",
+            definition: "TU*HR(-1)",
```

Comment:
This unit is not dimensionally compatible with the TestPhenomenon so this is not technically a valid unit definition.